File: go/raindrops/raindrops.go
package raindrops

import "strconv"

func Convert(number int) string {
	rainString := ""

	if number%3 != 0 && number%5 != 0 && number%7 != 0 {
		return strconv.Itoa(number)
	}
	if number%3 == 0 {
		rainString += "Pling"
	}

	if number%5 == 0 {
		rainString += "Plang"
	}

	if number%7 == 0 {
		rainString += "Plong"
	}
	return rainString
}

File: c/raindrops/raindrops.c
#include "raindrops.h"
#include "stdio.h"
#include <string.h>

char *convert(char result[], int drops) {
  strcpy(result, "");
  if (drops % 3 == 0) {
    strcat(result, "Pling");
  }

  if (drops % 5 == 0) {
    strcat(result, "Plang");
  }

  if (drops % 7 == 0) {
    strcat(result, "Plong");
  }

  if (!strcmp(result, "")) {
    sprintf(result, "%d", drops);
  }

  return result;
}
